### PR TITLE
[MI-3354] Fix issue: repeat dropdown not opening in config modal

### LINKED ISSUE #365: BUG - Config modal "Repeat" selector is broken
**Summary**

In the "Schedule" tab of the standup config modal, when clicking the "Repeat" selector arrow, the select box does not open. I am able to reproduce this in the following scenarios:

- Installing the latest release from https://github.com/standup-raven/standup-raven/releases/tag/v3.3.1
- Running the program locally in development, importing the `react-rrule-generator` as usual
- Running the program locally, directly importing `build/index.js` from a local cloned copy of https://github.com/standup-raven/react-bootstrap-rrule-generator

Issue does not reproduce in the scenario:

- Running the program locally, directly importing `src/lib/index.js` from a local cloned copy of `react-bootstrap-rrule-generator`

Building the `react-bootstrap-rrule-generator` project results in no change to `build/index.js`, so it seems its contents are already be sync'd with the `src` folder. I'm thinking there is something in the build process that is making this issue happen.

I'm not sure what the core issue is, but the bug exists in the production build of the plugin. Any thoughts on the what could be the issue or fix for this?

---

**Screenshots**

Timezone selector works

Repeat selector does not

**Environment**
-  Mattermost version: 8.0
-  OS: Mac M1
-  Platform: browser
-  Browser (if applicable): chrome

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**
### Summary
In the config modal, the "Repeat" dropdown was not opening on MM version `v7.8.4`. We were using the [fork](https://github.com/standup-raven/react-bootstrap-rrule-generator) of a [package](https://github.com/Fafruch/react-rrule-generator) to render this "Repeat" dropdown component. That package was deprecated so we moved the code from that package into the standup raven code itself and fixed some CSS styling.

image
image

### Ticket
Fixes #365

### CODE REVIEW

File: webapp/.eslintrc.json [hunk 1]
```diff
@@ -88,7 +88,7 @@
         "max-nested-callbacks": [2, {"max":2}],
         "max-statements-per-line": [2, {"max": 1}],
         "multiline-ternary": [1, "never"],
-        "new-cap": 2,
+        "new-cap": 0,
```
Comment (ayusht2810) [inline webapp/.eslintrc.json:91]: isn't the default value `0`?

File: webapp/src/components/rRule/rRule.jsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
-import RRuleGenerator from 'react-rrule-generator';
+import RRuleGenerator from '../reactBootstrapRRuleGenerator/index';
```
Comment (ayusht2810) [inline webapp/src/components/rRule/rRule.jsx:3]: do we need to write the `index` in import?

File: webapp/src/components/reactBootstrapRRuleGenerator/components/End/After.jsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import PropTypes from 'prop-types';
```
Comment (ayusht2810) [inline webapp/src/components/reactBootstrapRRuleGenerator/components/End/After.jsx:2]:
```suggestion

import PropTypes from 'prop-types';
```

File: webapp/src/components/reactBootstrapRRuleGenerator/components/End/OnDate.jsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import PropTypes from 'prop-types';
```
Comment (ayusht2810) [inline webapp/src/components/reactBootstrapRRuleGenerator/components/End/OnDate.jsx:2]: same

File: webapp/src/components/reactBootstrapRRuleGenerator/components/End/OnDate.jsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+import moment from 'moment';
+import DateTime from 'react-datetime';
```
Comment (ayusht2810) [inline webapp/src/components/reactBootstrapRRuleGenerator/components/End/OnDate.jsx:4]: fix import order. fix everywhere if fixing

File: webapp/src/components/reactBootstrapRRuleGenerator/components/ReactRRuleGenerator.jsx [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+class ReactRRuleGenerator extends React.Component {
+    // compute default view based on user's config
+    state = configureInitialState(
+        this.props.config,
+        this.props.calendarComponent,
+        this.props.id,
+    );
+
+    UNSAFE_componentWillMount() {
+        console.log('unmounting');
```
Comment (ayusht2810) [inline webapp/src/components/reactBootstrapRRuleGenerator/components/ReactRRuleGenerator.jsx:24]:
```suggestion
```

File: webapp/src/components/reactBootstrapRRuleGenerator/components/ReactRRuleGenerator.jsx [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+                repeat,
+                end,
+                options,
+                error,
+            },
+        } = this.state;
+
+        console.log('root root root root root root root ');
+        console.log(this.props.repeatDropdownStyle);
+        console.log('root root root root root root root ');
```
Comment (ayusht2810) [inline webapp/src/components/reactBootstrapRRuleGenerator/components/ReactRRuleGenerator.jsx:68]:
```suggestion
```

File: webapp/src/components/reactBootstrapRRuleGenerator/components/ReactRRuleGenerator.jsx [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+
+        return (
+            <div>
```
Comment (ayusht2810) [inline webapp/src/components/reactBootstrapRRuleGenerator/components/ReactRRuleGenerator.jsx:71]:
```suggestion
            <>
```

File: webapp/src/components/reactBootstrapRRuleGenerator/components/Repeat/Daily/index.jsx [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+                aria-label='Repeat daily interval'
+                className='form-control'
+                value={interval}
+                onChange={numericalFieldHandler(handleChange)}
+            />
+        </div>
+        <div className='col-sm-1'>
+            {translateLabel(translations, 'repeat.daily.days')}
+        </div>
+
```
Comment (ayusht2810) [inline webapp/src/components/reactBootstrapRRuleGenerator/components/Repeat/Daily/index.jsx:31]:
```suggestion
```

File: webapp/src/components/reactBootstrapRRuleGenerator/components/Repeat/Monthly/On.jsx [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+    handleChange,
+    translations,
+    monthlyOnDayDropdownStyle,
+}) => {
+    const isActive = mode === 'on';
+    let selected = on.day;
+
+    return (
+        <div className={`form-group d-flex align-items-sm-center text-bold ${!isActive && 'opacity-50'}`}>
+            <div className={'combo-label'}>
```
Comment (ayusht2810) [inline webapp/src/components/reactBootstrapRRuleGenerator/components/Repeat/Monthly/On.jsx:21]:
```suggestion
            <div className='combo-label'>
```

File: webapp/src/components/reactBootstrapRRuleGenerator/components/Repeat/Monthly/On.jsx [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+                {/*  id={`${id}-day`}*/}
+                {/*  name="repeat.monthly.on.day"*/}
+                {/*  aria-label="Repeat monthly on a day"*/}
+                {/*  className="form-control"*/}
+                {/*  value={on.day}*/}
+                {/*  disabled={!isActive}*/}
+                {/*  onChange={numericalFieldHandler(handleChange)}*/}
+                {/*>*/}
+                {/*  {[...new Array(31)].map((day, i) => <option key={i} value={i + 1}>{i + 1}</option>)}*/}
+                {/*</select>*/}
```
Comment (ayusht2810) [inline webapp/src/components/reactBootstrapRRuleGenerator/components/Repeat/Monthly/On.jsx:51]: remove this

File: webapp/src/components/reactBootstrapRRuleGenerator/components/Repeat/Monthly/On.jsx [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+
+                <SplitButton
+                    bsStyle={'link'}
```
Comment (ayusht2810) [inline webapp/src/components/reactBootstrapRRuleGenerator/components/Repeat/Monthly/On.jsx:54]:
```suggestion
                    bsStyle='link'
```

File: webapp/src/components/reactBootstrapRRuleGenerator/components/Repeat/Monthly/On.jsx [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+                    title={selected}
+                    disabled={!isActive}
+                    onSelect={(eventKey) => {
+                        selected = eventKey;
+                        return numericalFieldHandler(handleChange)({target: {name: 'repeat.monthly.on.day', value: eventKey}});
+                    }}
+                    style={monthlyOnDayDropdownStyle}
+                    dropup={true}
+                >
+                    {/*{[...new Array(31)].map((day, i) => <option key={i} value={i + 1}>{i + 1}</option>)}*/}
```
Comment (ayusht2810) [inline webapp/src/components/reactBootstrapRRuleGenerator/components/Repeat/Monthly/On.jsx:68]: same

File: webapp/src/components/reactBootstrapRRuleGenerator/components/Repeat/Monthly/OnThe.jsx [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+                    )}
+                </div>
+                <div className='col-sm-8'>
+                    {translateLabel(translations, 'repeat.monthly.on_the')}
+                </div>
+            </div>
+
+            <div className='week-of-month-selector'>
+                <SplitButton
+                    bsStyle={'link'}
```
Comment (ayusht2810) [inline webapp/src/components/reactBootstrapRRuleGenerator/components/Repeat/Monthly/OnThe.jsx:42]:
```suggestion
                    bsStyle='link'
```

Comment (ayusht2810) [reply webapp/src/components/reactBootstrapRRuleGenerator/components/Repeat/Monthly/OnThe.jsx:42]: fix everywhere

File: webapp/src/components/reactBootstrapRRuleGenerator/components/Repeat/Weekly/index.jsx [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+    translations,
+    weeklyFrequencyInputStyle,
+}) => {
+    let daysArray = toPairs(days);
+    if (options.weekStartsOnSunday) {
+        daysArray = daysArray.slice(-1).concat(daysArray.slice(0, -1));
+    }
+
+    return (
+        <div>
```
Comment (ayusht2810) [inline webapp/src/components/reactBootstrapRRuleGenerator/components/Repeat/Weekly/index.jsx:24]:
```suggestion
        <>
```

File: webapp/src/components/reactBootstrapRRuleGenerator/components/Repeat/Yearly/On.jsx [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+    hasMoreModes,
+    handleChange,
+    translations,
+}) => {
+    const daysInMonth = moment(on.month, 'MMM').daysInMonth();
+    const isActive = mode === 'on';
+
+    return (
+        <div className={`form-group row d-flex align-items-sm-center ${!isActive && 'opacity-50'}`}>
+            <div className='col-sm-1  '>
```
Comment (ayusht2810) [inline webapp/src/components/reactBootstrapRRuleGenerator/components/Repeat/Yearly/On.jsx:23]:
```suggestion
            <div className='col-sm-1'>
```

File: webapp/src/components/reactBootstrapRRuleGenerator/components/Repeat/Yearly/index.jsx [hunk 17]
```diff
@@ -0,0 +1,58 @@
...
+        onThe,
+        options,
+    },
+    handleChange,
+    translations,
+}) => {
+    const isTheOnlyOneMode = (option) => options.modes === option;
+    const isOptionAvailable = (option) => !options.modes || isTheOnlyOneMode(option);
+    return (
+        <div>
```
Comment (ayusht2810) [inline webapp/src/components/reactBootstrapRRuleGenerator/components/Repeat/Yearly/index.jsx:20]:
```suggestion
        <>
```

Comment (ayusht2810) [reply webapp/src/components/reactBootstrapRRuleGenerator/components/Repeat/Yearly/index.jsx:20]: fix everywhere

File: webapp/src/components/reactBootstrapRRuleGenerator/styles/react-datetime.css [hunk 18]
```diff
@@ -0,0 +1,230 @@
+/*!
+ * https://github.com/YouCanBookMe/react-datetime
```
Comment (ayusht2810) [inline webapp/src/components/reactBootstrapRRuleGenerator/styles/react-datetime.css:2]: use of this?

File: webapp/package-lock.json [hunk 19]
```diff
@@ -0,0 +1,10713 @@
+{
```
Comment (ayusht2810) [inline webapp/package-lock.json:1]: remove this file

**Review by ayusht2810 — DISMISSED** [review]

File: webapp/src/components/reactBootstrapRRuleGenerator/components/Repeat/Monthly/On.jsx [hunk 20]
```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import numericalFieldHandler from '../../../utils/numericalFieldHandler';
+import translateLabel from '../../../utils/translateLabel';
+import {MenuItem, SplitButton} from 'react-bootstrap';
```
Comment (avas27JTG) [inline webapp/src/components/reactBootstrapRRuleGenerator/components/Repeat/Monthly/On.jsx:5]:
```suggestion
import React from 'react';
import {MenuItem, SplitButton} from 'react-bootstrap';
import PropTypes from 'prop-types';

import numericalFieldHandler from '../../../utils/numericalFieldHandler';
import translateLabel from '../../../utils/translateLabel';
```

imports from node modules should be at top of custom imports

**Review by avas27JTG — APPROVED** [review]

**Review by avas27JTG — APPROVED** [review]

**Review by ayusht2810 — APPROVED** [review]

**Review by anmolghosh — APPROVED** [review]

**Pull request merged**